lib/ReactViews/Generic/TooltipWrapper.tsx

```diff
@@ -0,0 +1,306 @@
+/**
+ * base tooltipwrapperraw repurposed from magda, with some a11y modifications
+ */
+import React from "react";
+import ReactDOM from "react-dom";
+import { withTheme, DefaultTheme } from "styled-components";
+import { useUID } from "react-uid";
+const Box: any = require("../../Styled/Box").default;
+const TextSpan: any = require("../../Styled/Text").TextSpan;
+const RawButton: any = require("../../Styled/Button").RawButton;
+
+type Props = {
+  theme: DefaultTheme;
+
+  /** Invoked when the tooltip is dismissed by the user - not called if the tooltip disappears automatically  */
+  onDismiss?: () => void;
+  /**
+   * When dismiss/launch actions are handled within render prop, use this to disable default listeners
+   * if using this, ensure you handle the dismiss case externally otherwise the tooltip can get stuck on open
+   */
+  disableEventListeners?: boolean;
+  /** Whether the tooltip should start in an open state */
+  startOpen?: boolean;
+  /** Whether the tooltip should show up above or below the element it wraps */
+  orientation?: "below" | "above";
+  /**
+   * A function that returns a component that should launch the tooltip. If startOpen is false then it will show
+   * the tooltip when hovered over. Accepts an "open" callback that can be used to force-launch the tooltip (e.g. on click) */
+  launcher?: (launchObj: {
+    state: State;
+    launch: () => void;
+    forceSetState: (bool?: boolean) => void;
+  }) => React.ReactNode;
+  /** Styles to apply to the  actual tooltip */
+  innerElementStyles?: Object;
+  /** The tooltip content itself, as higher-order function that provides a function to dismiss the tooltip */
+  children: (applyAriaId: boolean, dismiss: () => void) => React.ReactNode;
+};
+
+type State = {
+  offset: number;
+  open: boolean;
+};
+
+/**
+ * @description Return a information tooltip, on hover show calculation method.
+ */
+class TooltipWrapperRaw extends React.Component<Props, State> {
+  rootRef = React.createRef<HTMLDivElement>();
+  tooltipTextElementRef = React.createRef<HTMLSpanElement>();
+  state = {
+    offset: 0,
+    open: !!this.props.startOpen
+  };
+
+  componentDidMount() {
+    if (!this.props.disableEventListeners) {
+      document.addEventListener("mousedown", this.dismiss);
+      document.addEventListener("touchstart", this.dismiss);
+    }
+
+    this.adjustOffset();
+  }
+
+  componentDidUpdate() {
+    this.adjustOffset();
+  }
+
+  componentWillUnmount() {
+    if (!this.props.disableEventListeners) {
+      document.removeEventListener("mousedown", this.dismiss);
+      document.addEventListener("touchstart", this.dismiss);
```

Comment:
Should this be `removeEventListener("touchstart", ...)`?

Reply:
Good catch, yes - this particular bit was straight from magda so guess they have something to fix too 😆